restservice/src/main/java/org/itevents/service/transactional/MyBatisUserService.java

```diff
@@ -137,4 +150,43 @@ public void checkPassword(User user, String password) {
             throw new WrongPasswordServiceException(message);
         }
     }
-}
+
+    @Override
+    public void setOtpToUser(User user, OtpGenerator otpGenerator) {
+        userDao.setOtpToUser(user, otpGenerator);
+    }
+
+    @Override
+    public User getUserByOtp(OtpGenerator otp) {
+        try {
+            return userDao.getUserByOtp(otp);
+        } catch (EntityNotFoundDaoException e) {
+            LOGGER.error(e.getMessage());
+            throw new EntityNotFoundServiceException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void sendEmailWithActivationLink(User user) throws Exception {
```

Comment:
Here we have violation of *Single responsibility principle*. From the mthod's name we assume that it sends email, but in fact it:
* create OTP;
* set OTP to user;
* build HTML for email;
* send email;